### PR TITLE
Dynamically render available styles for a product

### PR DESCRIPTION
**Pull request by shawnsull3**
Get Styles data from API, dynamically render thumbnails for each and renders price and style name based on the user selected style. 
url to trello story: https://trello.com/c/cA1BMYxv

Still need to limit only 4 styles per row, which I'll fix tomorrow while working on the sale price ticket.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -46,6 +47,7 @@ const mapDispatchToProps = (dispatch) => ({
   getReviewMetaData: () => dispatch(getReviewMetaData()),
   getProductData: () => dispatch(getProductData()),
   getProductsList: () => dispatch(getProductsList()),
+  getProductStyles: () => dispatch(getProductStyles()),
```
Comment (shawnsull3) [inline src/App.jsx:50]: Grabbing data from the API and storing it in the redux store

File: src/Components/ProductOverview/DescriptionFeaturesShare/DescriptionFeaturesShare.jsx [hunk 2]
```diff
@@ -1,11 +1,10 @@
 /* eslint-disable react/no-array-index-key */
 /* eslint-disable consistent-return */
-import React, { Component } from 'react';
```
Comment (shawnsull3) [inline src/Components/ProductOverview/DescriptionFeaturesShare/DescriptionFeaturesShare.jsx:3]: Remove to limit warnings in console

File: src/Components/ProductOverview/ProductInfo/ProductInfo.jsx [hunk 3]
```diff
@@ -43,6 +44,10 @@ const ProductInfo = ({ productData }) => (
 const mapStateToProps = (state) => ({
   reviewMetaData: state.reviewMetaReducer.reviewMetaData,
   productData: state.productDataReducer.productData,
+  productStyles: state.productDataReducer.productStyles,
+  selectedStyle_id: state.productDataReducer.selectedStyle_id,
+  selectedPrice: state.productDataReducer.selectedPrice,
+  salePrice: state.productDataReducer.salePrice,
```
Comment (shawnsull3) [inline src/Components/ProductOverview/ProductInfo/ProductInfo.jsx:50]: Mapping all this API data to props

File: src/Components/ProductOverview/ProductInfo/Styles.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <div>
+      <div className="style-name">Style: {props.selectedStyleName}</div>
+      {props.productStyles.results.map((style, index) => (
+        <img
+          src={style.photos[0].thumbnail_url}
+          alt={style.name}
+          id={index}
+          className="rounded-circle style-pics"
+          key={style.style_id}
+          onClick={() => props.updateSelectedStyle(props.productStyles.results[index])}
```
Comment (shawnsull3) [inline src/Components/ProductOverview/ProductInfo/Styles.jsx:21]: Couldn't use destructuring or this onClick event would not work. Therefore, ignored destructuring eslint.

File: src/Components/ProductOverview/ProductInfo/Styles.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        />
+      ))}
+    </div>
+  ) : <div />
```
Comment (shawnsull3) [inline src/Components/ProductOverview/ProductInfo/Styles.jsx:25]: Renders all styles available for the currently selected product. Uses the first thumbnail in the photos array.

File: src/Components/ProductOverview/ProductOverview.css [hunk 6]
```diff
@@ -50,9 +50,30 @@
...
+
 .slogan {
     font-size: 1.1em;
     color: #323639;
     font-weight: bold;
 }
 
+.style-pics {
+    /* background-color: #323639; */
+    object-fit: cover;
```
Comment (shawnsull3) [inline src/Components/ProductOverview/ProductOverview.css:67]: Fixes image distortion

File: src/Components/ProductOverview/ProductOverview.css [hunk 7]
```diff
@@ -50,9 +50,30 @@
...
+    /* width: 2.7em;
+    height: 2.7em; */
+    width: 50px;
+    height: 50px;
```
Comment (shawnsull3) [inline src/Components/ProductOverview/ProductOverview.css:71]: Unsure which to use currently for the thumbnail styling. Any thoughts?

File: src/Components/ProductOverview/ProductOverview.jsx [hunk 8]
```diff
@@ -1,24 +1,44 @@
...
-const ProductOverview = () => (
-  <div className="product-overview">
-    <div className="container-fluid">
-      <div className="row">
-        <div className="border col-12 col-sm-8 col-lg-8 col-xl-8 image-gallery">
-          ImageGallery
-        </div>
-        <div className="border col-12 col-sm-4 col-lg-4 col-xl-4 product-info">
-          <ProductInfo />
+class ProductOverview extends Component {
```
Comment (shawnsull3) [inline src/Components/ProductOverview/ProductOverview.jsx:7]: May need this component to be a class later, so I refactored it.

File: src/Shared/HelperFunctions.js [hunk 9]
```diff
@@ -14,4 +14,12 @@ const calcAvgRating = (ratings) => {
...
-export default calcAvgRating;
+export const defaultStyleFinder = (stylesArr) => {
+  for (let i = 0; i < stylesArr.length; i++) {
+    if (stylesArr[i]['default?'] === 1) {
+      return stylesArr[i];
+    } else {
+      return stylesArr[0];
+    }
+  }
+};
```
Comment (shawnsull3) [inline src/Shared/HelperFunctions.js:25]: This helper function iterates through the styles.result array received from the API and determines which style is the default, if no default define it sets the first style as the default.

File: src/data/actions/productsListAction.jsx [hunk 10]
```diff
@@ -4,10 +4,7 @@ import axios from 'axios';
 import { GET_PRODUCTS_LIST_SUCCESS } from '../types/types';
 
 export const getProductsList = () => (dispatch) => axios.get('http://3.134.102.30/products/list')
-  .then((res) => {
-    console.log(res.data);
```
Comment (shawnsull3) [inline src/data/actions/productsListAction.jsx:8]: removed unneeded console.log

File: src/data/reducers/productDataReducer.jsx [hunk 11]
```diff
@@ -11,6 +16,19 @@ const productDataReducer = (state = initialState, action) => {
...
+        productStyles: action.payload,
+      };
+    case UPDATE_SELECTED_STYLE:
+      return {
+        ...state,
+        selectedStyle_id: action.payload.style_id,
+        selectedPrice: action.payload.original_price,
+        salePrice: action.payload.sale_price,
+        selectedStyleName: action.payload.name,
+      };
```
Comment (shawnsull3) [inline src/data/reducers/productDataReducer.jsx:31]: reducer for updating data in the redux base on the user selected style

**Review by morahan — APPROVED** [review]

**Pull request merged**